statsmodels/genmod/generalized_linear_model.py

```diff
@@ -607,9 +607,15 @@ def hessian(self, params, scale=None, observed=None):
             else:
                 observed = True
 
+        try:
+            hess
+        except NameError:
+            hess = np.empty_like(self.exog)
+
         factor = self.hessian_factor(params, scale=scale, observed=observed)
-        hess = -np.dot(self.exog.T * factor, self.exog)
-        return hess
+        hess = np.empty_like(self.exog)
```

Comment:
this line can be deleted, hess is already created, and this needs to allocate the memory again